src/main/java/net/dries007/tfc/common/blocks/devices/PitKilnBlock.java

```diff
@@ -207,9 +217,10 @@ public ItemStack getCloneItemStack(BlockState state, HitResult result, LevelRead
         return ItemStack.EMPTY;
     }
 
+    @SuppressWarnings("deprecation")
     @Override
-    protected void beforeRemove(InventoryBlockEntity<?> entity)
+    protected float getDestroyProgress(BlockState state, Player player, BlockGetter level, BlockPos pos)
     {
-        // No-op - we don't want to call ejectInventory()
+        return state.getValue(STAGE) == LIT ? 0F : super.getDestroyProgress(state, player, level, pos);
```

Comment:
I don't understand why a lit pit kiln should be unbreakable. What if you decided you didn't want to burn it anymore? What if you break the blocks around the pit kiln thus invalidating it?